### PR TITLE
Added powershell version to Metadata of Template

### PR DESCRIPTION
**Pull request by SomilGanguly**
## Overview/Summary

This Pull Request is for merging the Metadata Enhancements to AzOps. 

## This PR fixes/adds/changes/removes

1. Adds - Metadata to the AzOps pull templates

### Breaking Changes

1. N/A

## Testing Evidence

This feature has been tested and verified.

image

## As part of this Pull Request I have

- [ ] Checked for duplicate [Pull Requests](https://github.com/Azure/azops/pulls)
- [ ] Associated it with relevant [issues](https://github.com/Azure/azops/issues), for tracking and closure.
- [ ] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/azops/tree/main)
- [ ] Performed testing and provided evidence.
- [ ] Updated relevant and associated documentation.
- [ ] Updated the ["What's New?"](https://github.com/Azure/Enterprise-Scale/wiki/Whats-new) wiki page (located in the [Enterprise-Scale repo](https://github.com/Azure/Enterprise-Scale) in the directory: `/docs/wiki/whats-new.md`)

### CODE REVIEW

**Comment by uday31in** [comment]
I very much like the idea however i worry about churn it will generate when PowerShell module is updated. Thoughts?

**Comment by daltondhcp** [comment]
Agree with @uday31in (and as was also mentioned in #363 related to this feature) - to reduce potential template churn I'd prefer to include only name and not version in the metadata.

File: src/data/template/Microsoft.Authorization/policyAssignments.template.jq [hunk 1]
```diff
@@ -1,6 +1,12 @@
 {
     "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
     "contentVersion": "1.0.0.0",
+    "metadata": {
+        "_generated": {
```
Comment (daltondhcp) [inline src/data/template/Microsoft.Authorization/policyAssignments.template.jq:5]: To align with the schema used in [bicep](https://github.com/Azure/bicep/issues/800), the property should be `_generator` and not `_generated`. Please change that for all instances. 

image

**Comment by SomilGanguly** [comment]
> I very much like the idea however i worry about churn it will generate when PowerShell module is updated. Thoughts?

Agreed Uday, So as per Johan's comment have removed the version and updated the metadata name to _generator

File: src/data/template/Microsoft.Authorization/policyAssignments.template.jq (hunk 1 above)
Comment (SomilGanguly) [reply src/data/template/Microsoft.Authorization/policyAssignments.template.jq]: > To align with the schema used in [bicep](https://github.com/Azure/bicep/issues/800), the property should be `_generator` and not `_generated`. Please change that for all instances.
> 
> image
Agreed Johan, have removed the version and updated the metadata name to _generator.

**Review by daltondhcp — APPROVED** [review]
LGTM

**Pull request merged**